disgover/disgover_transport_grpc.go

```diff
@@ -154,10 +218,106 @@ func (this *DisGoverService) peerUpdateGrpc() {
 	}
 }
 
+// UpdateSoftwareGrpc
+func (this *DisGoverService) UpdateSoftwareGrpc(ctx context.Context, softwareUpdate *proto.SoftwareUpdate) (*proto.Empty, error) {
+
+	// Verify seed node is authentic?
+	err := this.verifySeedNode(softwareUpdate.Authentication)
+	if err != nil {
+		return &proto.Empty{}, err
+	}
+
+	// Write file.
+	fileName :=  "." + string(os.PathSeparator) + "disgo"
+	err = ioutil.WriteFile(fileName, softwareUpdate.Software, 0)
+	if err != nil {
+		utils.Error(fmt.Sprintf("unable to save file %s", fileName), err)
+		return &proto.Empty{}, err
+	}
+
+	utils.Info(fmt.Sprintf("software updated from seed node"))
+
+	go func() {
+		time.Sleep(3 * time.Second) // TODO: This should be set to the time from the seed!!!
+		services.GetDbService().Close()
+		utils.Info("rebooting with new version of disgo...")
+		os.Exit(0)
+	}()
+
+	return &proto.Empty{}, nil
+}
+
+// peerUpdateSoftwareGrpc
+func (this *DisGoverService) peerUpdateSoftwareGrpc(software []byte) {
+
+	// Get delegates in cache.
+	delegates, err := types.ToNodesByTypeFromCache(services.GetCache(), types.TypeDelegate)
+	if err != nil {
+		utils.Error(err)
+		return
+	}
+
+	// New authentication.
+	authentication, err := types.NewAuthenticate()
+	if err != nil {
+		utils.Error(err)
+		return
+	}
+
+	for _, delegate := range delegates {
+		maxSize := 1024 * 1024 * 1024 // TODO: Do we need this (ServerOptions sets this)?
+		conn, err := grpc.Dial(fmt.Sprintf("%s:%d", delegate.GrpcEndpoint.Host, delegate.GrpcEndpoint.Port), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxSize), grpc.MaxCallSendMsgSize(maxSize)), grpc.WithInsecure())
+		if err != nil {
+			utils.Warn(fmt.Sprintf("cannot dial node [host=%s, port=%d]", delegate.GrpcEndpoint.Host, delegate.GrpcEndpoint.Port), err)
+			continue
+		}
+		client := proto.NewDisgoverGrpcClient(conn)
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+
+		// Update software.
+		_, err = client.UpdateSoftwareGrpc(ctx, &proto.SoftwareUpdate{Authentication: convertToProtoAuthentication(authentication), Software: software})
+		if err != nil {
+			utils.Warn(fmt.Sprintf("unable to update sofware [address=%s, host=%s, port=%d]", delegate.Address, delegate.GrpcEndpoint.Host, delegate.GrpcEndpoint.Port))
+			continue
+		}
+		conn.Close()
+		cancel()
+	}
+}
+
+// verifySeedNode
+func (this *DisGoverService) verifySeedNode(protoAuthenticate *proto.Authentication) error {
+
+	if protoAuthenticate == nil {
+		utils.Warn("attempted update by a non-authorized seed node")
+		return errors.New("you are not an authorized seed node")
+	}
+
+	authentication := convertToDomainAuthentication(protoAuthenticate)
+	authenticationAddress, err := authentication.GetAddress()
+	if err != nil {
+		utils.Error(err)
+		return err
+	}
+
+	for _, seedAddress := range types.GetConfig().SeedAddresses {
```

Comment:
Rather than add a new array to the config, why not just use the Node.address field?
E.g.; use this:
```for _, seedEndpoint := range types.GetConfig().SeedEndpoints {
        if seedEndpoint.address == authenticationAddress {```